src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,801 @@
+      case this.secondEl:
+        switch (event.key) {
+          case "ArrowLeft":
+            this.setFocus("minute");
+            break;
+          case "ArrowRight":
+            if (this.hourDisplayFormat === "12") {
+              this.setFocus("meridiem");
+            }
+            break;
+        }
+        break;
+      case this.meridiemEl:
+        switch (event.key) {
+          case "ArrowLeft":
+            if (this.step !== 60) {
+              this.setFocus("second");
+            } else {
+              this.setFocus("minute");
+            }
+            break;
+        }
+        break;
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  @Method()
+  async setFocus(target: TimeComponents): Promise<void> {
+    this[`${target || "hour"}El`].focus();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private decrementHour = (event?: KeyboardEvent | MouseEvent): void => {
+    if (event && event instanceof KeyboardEvent && event.key !== "Enter") {
+      return;
+    }
+    const newHour = !this.hour ? 0 : this.hour === "00" ? 23 : parseInt(this.hour) - 1;
+    this.setTime("hour", newHour);
+  };
+
+  private decrementMeridiem = (event?: KeyboardEvent | MouseEvent): void => {
+    if (event && event instanceof KeyboardEvent && event.key !== "Enter") {
+      return;
+    }
+    const newMeridiem = this.meridiem === "PM" ? "AM" : "PM";
+    this.setTime("meridiem", newMeridiem);
+  };
+
+  private decrementMinuteOrSecond = (key: MinuteOrSecond): void => {
+    let newValue;
+    if (stringIsValidNumber(this[key])) {
+      const valueAsNumber = parseInt(this[key]);
+      if (valueAsNumber === 0) {
+        newValue = 59;
+      } else {
+        newValue = valueAsNumber - 1;
+      }
+    } else {
+      newValue = 59;
+    }
+    this.setTime(key, newValue);
+  };
+
+  private decrementMinute = (event?: KeyboardEvent | MouseEvent): void => {
+    if (event && event instanceof KeyboardEvent && event.key !== "Enter") {
+      return;
+    }
+    this.decrementMinuteOrSecond("minute");
+  };
+
+  private decrementSecond = (event?: KeyboardEvent | MouseEvent): void => {
+    if (event && event instanceof KeyboardEvent && event.key !== "Enter") {
+      return;
+    }
+    this.decrementMinuteOrSecond("second");
+  };
+
+  private focusHandler = (event: FocusEvent): void => {
+    this.activeEl = event.target as HTMLSpanElement;
+  };
+
+  private getMeridiem = (): Meridiem => {
+    if (stringIsValidNumber(this.hour)) {
+      const hourAsNumber = parseInt(this.hour);
+      return hourAsNumber >= 0 && hourAsNumber <= 11 ? "AM" : "PM";
+    }
+    return null;
+  };
+
+  private getDisplayHour(): string {
```

Comment:
nitpick: maybe some of these functions could be moved to a utility to keep logic separate?

Reply:
I moved `getMeridiem` to the time util and created a `getMeridiemHour` util that extracts all the necessary logic that `getDisplayHour` needs to get the correct display hour.